### PR TITLE
Alerts constants

### PR DESCRIPTION
**Pull request by ekoyle**
Use some constants instead of magic numbers to help with readability of alerting code.

### CODE REVIEW

**Comment by laf** [comment]
Running on my home install to test.

**Comment by laf** [comment]
Works ok. Worth fixing the scrut issue though:

https://scrutinizer-ci.com/g/librenms/librenms/inspections/25c206d7-96ed-4a4c-9462-5a19f66369b8/issues/files/includes/alerts.inc.php?status=new&orderField=path&order=asc&honorSelectedPaths=0

File: alerts.php [hunk 1]
```diff
@@ -154,7 +155,7 @@ function RunAcks() {
         }
 
         $alert['details'] = json_decode(gzuncompress($alert['details']), true);
-        $alert['state']   = 2;
+        $alert['state']   = AlertState::WORSE;
```
Comment (f0o) [inline alerts.php]: This is not 'worse' but 'ack'

File: alerts.php [hunk 2]
```diff
@@ -234,12 +237,15 @@ function RunAlerts() {
...
+        // noiss - do not issue alert
         $noiss            = false;
+        // noacc - do not close alert ?
         $noacc            = false;
+        // updet - update details in alert_log ?
         $updet            = false;
         $rextra           = json_decode($alert['extra'], true);
         $chk              = dbFetchRow('SELECT alerts.alerted,devices.ignore,devices.disabled FROM alerts,devices WHERE alerts.device_id = ? && devices.device_id = alerts.device_id && alerts.rule_id = ?', array($alert['device_id'], $alert['rule_id']));
-        if ($chk['alerted'] == $alert['state']) {
+        if ($chk['alerted'] == $alert['state']) { // FIXME: this could be made more clear, should probably also be double-checked
```
Comment (f0o) [inline alerts.php]: If the accounted state equals the current state, dont issue alert

Comment (ekoyle) [inline alerts.php]: I guess my confusion is on the 'alerted' field in the database.  It sounds like a boolean from the name, is it actually the most recently alerted state?

**Comment by ekoyle** [comment]
@laf: I think those lines scrutinizer is complaining about are doing what is intended.  I could maybe change it to ($r === false || $r === 0 || $r === null) to be more explicit?

dbUpdate will either return mysqli?_affected_rows() or false.  I think this was intended to match on either 0 or false or null (I'm assuming null is possible if there is an unanticipated error?).

I stored the return value to do some debugging, but then got rid of all the debugging.  I can get rid of the reference to $r if you'd like.

File: alerts.php (hunk 2 above)
Comment (f0o) [inline alerts.php]: for the delay, yeah it's the state that was sent out last.

File: html/includes/reports/alert-log.pdf.inc.php [hunk 3]
```diff
@@ -41,24 +43,24 @@
     $hostname    = gethostbyid(mres($alert_entry['device_id']));
     $alert_state = $alert_entry['state'];
 
-    if ($alert_state != '') {
-        if ($alert_state == '0') {
+    if ($alert_state != '') { // FIXME: would is_numeric($alert_state) be more appropriate here?
```
Comment (ekoyle) [inline html/includes/reports/alert-log.pdf.inc.php]: @f0o: what do you think on this one?  Could $alert_entry['state'] be null?  If so, should it be treated the same as 0?

**Comment by laf** [comment]
Scrut needs looking at:

https://scrutinizer-ci.com/g/librenms/librenms/inspections/a08f099f-37c2-4155-a6aa-cada45487ab5/issues/files/includes/alerts.inc.php?status=new&orderField=path&order=asc&honorSelectedPaths=0

**Comment by laf** [comment]
@f0o any further thoughts on this one? I'm happy to update my home install with it again.

**Comment by laf** [comment]
bump

**Comment by ekoyle** [comment]
@f0o: are you interested in this, or should we just scrap it?

**Comment by ekoyle** [comment]
Okay, rebased again... can we remove the conflict label?

**Comment by laf** [comment]
@f0o any thoughts on this?

**Comment by ekoyle** [comment]
Should I keep updating this?

**Comment by laf** [comment]
I wouldn't say so just now. @f0o ?

**Comment by Rosiak** [comment]
Ping Mr. @f0o

**Comment by laf** [comment]
@f0o?

**Comment by laf** [comment]
I think it's best to close this, appreciate the effort in contributing @ekoyle. Sorry we've not been able to push this one forward.

**Pull request closed without merging**